share/spack/gitlab/cloud_pipelines/.gitlab-ci.yml

```diff
@@ -479,7 +479,7 @@ developer-tools-manylinux2014-build:
     SPACK_CI_STACK_NAME: developer-tools-darwin
 
 developer-tools-darwin-generate:
-  tags: [ "macos-ventura", "apple-clang-15", "aarch64-macos" ]
```

Comment:
What is the xcode installed now? Did it move to 16, if so should that be a tag?

We don't have any other Mac runners so it may not matter, but we lose information in the tag here.

Reply:
It is the latest xcode now, and it is apple-clang-16. The reason I removed it was, because the compilers are auto-detected and the version is not specified at all anywhere in the darwin stacks, what is the point? It seems like, either we care what compiler is on the system in which case we specify it in package preference and in the compilers.yaml, or we let spack auto-detect it and we just let it do that. What do you think?

Reply:
Following up-- do you think we should include `apple-clang-<version>` tagging in our scheme?

Reply:
OK, just updated this PR to include tagging for apple-clang version